snuba/consumer.py

```diff
@@ -77,10 +76,17 @@ def __init__(
             JSONRowEncoder(),
         )
 
+        self.__processor: MessageProcessor
         self.__pre_filter = table_writer.get_stream_loader().get_pre_filter()
-        self.__processor = (
-            self.__storage.get_table_writer().get_stream_loader().get_processor()
-        )
+
+    def get_processor(self) -> MessageProcessor:
+        try:
+            return self.__processor
+        except AttributeError:
+            self.__processor = (
+                self.__storage.get_table_writer().get_stream_loader().get_processor()
+            )
+            return self.__processor
```

Comment:
This is only public to allow the subclass to override it.

Ideally the processor (and writer, and filter) would just be passed in to the constructor to make this unnecessary but that requires changes to the builder that I don't want to make right now.

Reply:
Since this is public only to be overridden, could you please make it "protected" `_get_processor` so you will be able to override it and it says it should not be accessed externally.